oauth/oauth.go

```diff
@@ -28,7 +25,9 @@ type Config struct {
 }
 
 // NewConfig creates a new *Config for the provided client credentials.
-func NewConfig(clientID, clientSecret, refreshToken string) *Config {
+func NewConfig(clientID, clientSecret, refreshToken, apiTld string) *Config {
```

Comment:
This is a breaking change. I suggest to just do the following in `(*client).OAuthClient`:
```
func (c *Config) OAuthClient(ctx context.Context) *http.Client {
	oauthConfig := oauth.NewConfig(c.ClientID, c.ClientSecret, c.RefreshToken)
        if c.TokenURL != "" {
                oauthConfig.Endpoint.TokenURL = c.TokenURL
        }

	return oauthConfig.Client(ctx)
}
```

Reply:
Changed in https://github.com/Bonial-International-GmbH/site24x7-go/pull/34